### PR TITLE
chore(deps): update vite

### PR DESCRIPTION
**Pull request by hi-ogawa**
### Description

Related https://github.com/vitest-dev/vitest/pull/5123#issuecomment-1949788480, https://github.com/vitest-dev/vitest/pull/5193

I tweaked vite-node watch test cases with a quick workaround. I'll see how others go on CI.

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [ ] It's really useful if your PR references an issue where it is discussed ahead of time. If the feature is substantial or introduces breaking changes without a discussion, PR might be closed.
- [ ] Ideally, include a test that fails without this PR but passes with it.
- [ ] Please, don't make changes to `pnpm-lock.yaml` unless you introduce a new test example.

### Tests
- [ ] Run the tests with `pnpm test:ci`.

### Documentation
- [ ] If you introduce new functionality, document it. You can run documentation with `pnpm run docs` command.

### Changesets
- [x] Changes in changelog are generated from PR name. Please, make sure that it explains your changes in an understandable manner. Please, prefix changeset messages with `feat:`, `fix:`, `perf:`, `docs:`, or `chore:`.

### CODE REVIEW

**Comment by hi-ogawa** [comment]
Maybe flakiness increased or maybe it's just same as before. I retried three times and Windows CI hasn't go through yet.
https://github.com/vitest-dev/vitest/actions/runs/8106298865?pr=5311

**Comment by sheremet-va** [comment]
> Maybe flakiness increased or maybe it's just same as before. I retried three times and Windows CI hasn't go through yet.
> 
> https://github.com/vitest-dev/vitest/actions/runs/8106298865?pr=5311

Yea, it increased. I have another PR open with this change and it's not merged because of the flakiness. We need to rewrite the waitForWatcher to wait for a specific file. The watcher is ready too soon.

**Comment by hi-ogawa** [comment]
> The watcher emits ready event too soon.

@sheremet-va Do you know where this change is coming from? I couldn't find any specific change on Vite, so maybe something from chokidar (bumped to 3.6.0 in https://github.com/vitejs/vite/pull/15803)?

**Comment by sheremet-va** [comment]
> > The watcher emits ready event too soon.
> 
> @sheremet-va Do you know where this change is coming from? I couldn't find any specific change on Vite, so maybe something from chokidar (bumped to 3.6.0 in [vitejs/vite#15803](https://github.com/vitejs/vite/pull/15803))?

Could be anything, I don't know why this started happening, but watcher was always unstable.

**Comment by sheremet-va** [comment]
Maybe we can pass down a ENV variable and wait for it to appear in `watcher.getWatched()`? Like here: https://github.com/vitejs/vite/blob/63a39c244b08cf1f2299bc2c3cfddcb82070d05b/playground/hmr-ssr/__tests__/hmr.spec.ts#L971

**Comment by hi-ogawa** [comment]
Thanks for the pointer. I didn't know you were referring `watcher.getWatched()` when you were talking about vite runtime tests https://github.com/vitest-dev/vitest/pull/5123#issuecomment-1949921957.

For vite-node tests, we can wait for entry files to be included `watcher.getWatched()`, but I'm not sure what we should target for Vitest watch tests.

**Comment by sheremet-va** [comment]
> For vite-node tests, we can wait for entry files to be included `watcher.getWatched()`, but I'm not sure what we should target for Vitest watch tests.

We can use something lime this `VITEST_TEST_WATCHER_FILE=filename vitest run`

**Comment by hi-ogawa** [comment]
> We can use something like this VITEST_TEST_WATCHER_FILE=filename vitest run

Failing CI is indicating that `runVitestCli` is stuck before outputting `Waiting for file changes`:

 https://github.com/vitest-dev/vitest/actions/runs/8135325279/job/22229683329#step:10:977

https://github.com/vitest-dev/vitest/blob/c3eb8deb768f6ef1adff8e6644ce99de58fd8b01/test/test-utils/index.ts#L191-L196

So, I'm not sure if this flakiness can be fixed by waiting any specific file by `VITEST_TEST_WATCHER_FILE=filename`.

File: packages/vite-node/src/cli.ts [hunk 1]
```diff
@@ -131,6 +131,31 @@ async function run(files: string[], options: CliOptions = {}) {
     process.on('uncaughtException', (err) => {
       console.error(c.red('[vite-node] Failed to execute file: \n'), err)
     })
+
+    if (process.env.VITE_TEST_WATCHER_DEBUG) {
```
Comment (sheremet-va) [inline packages/vite-node/src/cli.ts:135]: Can we add `process.env.VITE_TEST_WATCHER_DEBUG: false` in `rollup` config (under a condition like "production" build) so this is removed when bundling?

**Comment by sheremet-va** [comment]
> Failing CI is indicating that `runVitestCli` is stuck before outputting `Waiting for file changes`:

This error happens more on Mac OS, to be honest. For some reason, this line is not reported at all.

**Comment by hi-ogawa** [comment]
> > Failing CI is indicating that `runVitestCli` is stuck before outputting `Waiting for file changes`:
> 
> This error happens more on Mac OS, to be honest. For some reason, this line is not reported at all.

While searching I noticed this one `VITE_TEST_WATCHER_DEBUG`, I noticed this darwin specific one:

https://github.com/vitest-dev/vitest/blob/dcc8dda2e2813a40671f864049e089f459766834/packages/vite-node/src/hmr/emitter.ts#L33-L44

Maybe tweak this to see if it helps anything?

File: packages/vite-node/src/cli.ts (hunk 1 above)
Comment (hi-ogawa) [reply packages/vite-node/src/cli.ts:135]: It looks like I can use `process.env.ROLLUP_WATCH` to remove this only on build https://github.com/vitest-dev/vitest/pull/5311/commits/17aa6951edecc12f89da0fb7ba636a77bded1ddc

Comment (hi-ogawa) [reply packages/vite-node/src/cli.ts:135]: Ah, this obviously breaks CI since it uses build... Let me think about a different way.

**Comment by sheremet-va** [comment]
@hi-ogawa do you think this is stable enough?

**Comment by hi-ogawa** [comment]
> do you think this is stable enough?

Ubuntu and macos CI seems okay. I cannot really judge Windows behavior, so I can only hope it's stable enough.

**Review by sheremet-va — APPROVED** [review]

**Pull request merged**